src/unaudited/examples/ECDSAStakeRegistryPermissioned.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import {ISignatureUtils} from "eigenlayer-contracts/src/contracts/interfaces/ISignatureUtils.sol";
+import {ECDSAStakeRegistry} from "../ECDSAStakeRegistry.sol";
+import {IDelegationManager} from "eigenlayer-contracts/src/contracts/interfaces/IDelegationManager.sol";
+
+/// @title ECDSA Stake Registry with an Operator Allowlist
+/// @notice This contract extends ECDSAStakeRegistry by adding functionality to allowlist and remove operators
+contract ECDSAStakeRegistryPermissioned is ECDSAStakeRegistry {
+    mapping(address => bool) public allowlistedOperators;
+
+    /// @dev Custom error to signal that an operator is not allowlisted.
+    error OperatorNotAllowlisted();
+
+    constructor(IDelegationManager _delegationManager) ECDSAStakeRegistry(_delegationManager) {
+        // _disableInitializers();
+    }
+
+    /// @notice Adds an operator to the allowlisted operator set
+    /// @dev An allowlisted operator isn't a part of the operator set. They must subsequently register themselves
+    /// @param _operator The address of the operator to be allowlisted
+    function permitOperator(address _operator) external onlyOwner {
+        allowlistedOperators[_operator] = true;
+    }
+
+    /// @notice Revokes an operator's permission and deregisters them
+    /// @param _operator The address of the operator to remove from the allowlist and deregistered.
+    function revokeOperator(address _operator) external onlyOwner {
+        delete allowlistedOperators[_operator];
+        if (_operatorRegistered[_operator]){
```

Comment:
think this check is redundant, `_deregisterOperator` seems to already perfom it

Reply:
We need this check; otherwise, we will revert to the check you're talking about if the operator isn't registered.  

We could probably have two functions, `revokeOperator` and `revokeAndEjectOperator`, so we could remove checking the value twice, but I think gas here isn't too important since it's an admin function